ddl/index.go

```diff
@@ -1509,17 +1516,43 @@ func (w *worker) updateReorgInfoForPartitions(t table.PartitionedTable, reorg *r
 	return false, errors.Trace(err)
 }
 
-func findIndexesByColName(indexes []*model.IndexInfo, colName string) ([]*model.IndexInfo, []int) {
-	idxInfos := make([]*model.IndexInfo, 0, len(indexes))
-	offsets := make([]int, 0, len(indexes))
-	for _, idxInfo := range indexes {
-		for i, c := range idxInfo.Columns {
-			if strings.EqualFold(colName, c.Name.L) {
-				idxInfos = append(idxInfos, idxInfo)
-				offsets = append(offsets, i)
+type indexesToChange struct {
+	indexInfo  *model.IndexInfo
+	isChanging bool // whether the index is created by modify column
```

Comment:
Name and meaning are not very appropriate.

Reply:
Do you have any suggestion?

Reply:
How about `isGenerated`?

Reply:
It seems easy to mix this concept up with "generated columns"...

We just need to mention "the index is in a middle(invisible) state".